### ISSUE TITLE
Cannot scroll on composer overlay when creating a new thread

### ISSUE THREAD

**Issue by johnnynotsolucky** [withspectrum/spectrum#3071]
Reported here: https://spectrum.chat/?t=3a8e24ab-972f-496b-946a-2812ebd7c94b

> Hey friends. If I have the "new conversation" modal open on my macbook 15", and I scroll down so the modal isn't visible anymore, then I can't scroll up anymore and get it back. 

I thought it might be an easy fix, https://github.com/withspectrum/spectrum/pull/3070, but I was wrong 😕

**Comment by brianlovin** [comment]
@johnnynotsolucky I think the optimal fix here is to refactor so that we only have one composer component. Right now we have two - the first is what you see if you compose a thread from the inbox, the second is what you see when composing from a community/channel profile view.

What might work is if we had the composer inside a 'slideout' component, similar to threads - that way it could be toggled from anywhere and would appear on top of whatever context you're in. What do you think?

**Comment by johnnynotsolucky** [comment]
Yeah I agree, I really like the slideout idea. It would also be nice to have the same composing experience from the inbox and profile view 🙂

**Comment by ThomasRoest** [comment]
you could remove the overlay temporarily, so you have normal scroll behaviour
Image 2018-12-15 at 4.14.16 PM

**Comment by ThomasRoest** [comment]
I've noticed a couple of issues concerning the composer design etc. @brianlovin this is an example of what a slide out composer could look like

<img width="1278" alt="image 2019-01-07 at 1 43 51 pm" src="https://user-images.githubusercontent.com/4428811/50769006-d6c82480-1282-11e9-9e32-7d6c69e51128.png">

**Comment by brianlovin** [comment]
@ThomasRoest I would love to move to this slideout composer, and ideally combine it with the dashboard thread composer as well (right now it's 2 components - tech debt). Your mock feels so much better - do you think it's feasible to combine?

**Comment by ThomasRoest** [comment]
I'm not sure yet ( considering the amount of work etc) But I'll look into it!

**Comment by ThomasRoest** [comment]
What's possible is adding the same composer from the community to the dashboard like this:

![screen recording 2019-01-15 at 03 40 pm](https://user-images.githubusercontent.com/4428811/51187595-779a8d80-18dc-11e9-952d-d0fae2c24d6d.gif)

**Comment by brianlovin** [comment]
This doesn't seem too bad (although we should attempt to stretch the contents of the composer so that the channel selection and publish button are at the bottom of the page and don't get pushed off if the thread becomes long.

Wdyt @superbryntendo ?

**Comment by superbryntendo** [comment]
totally agree. i would love to have the buttons pinned top or bottom and have the text area scroll.

**Comment by ThomasRoest** [comment]
in `/components/rich-text-editor/container.js` should I use the editor with version 2 prop?

` if (version === 2) {`

**Comment by superbryntendo** [comment]
if we're moving all of them to one component, we shouldn't need that prop anymore. iirc I used that to handle the switch to DraftJS.

**Comment by ThomasRoest** [comment]
ok, and what code block is more recent / should be leading? first or second?

**Comment by ThomasRoest** [comment]
there are a few issues left to fix, but this is the current status:

![newcomposer](https://user-images.githubusercontent.com/4428811/51425287-2ccd8e00-1bda-11e9-802a-a03205818304.gif)

**Comment by mxstbr** [comment]
Amazing @ThomasRoest! Want to submit a first PR and then continue iterating? That would help me figure out if I will block you with #4564
